.github/workflows/call-build-linux-arm-packages.yml

```diff
@@ -0,0 +1,78 @@
+name: Builds Linux arm binary packages into S3 bucket.
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: The version of Valkey to create.
+        type: string
+        required: true
+      ref:
+        description: The commit, tag or branch of Valkey to checkout for building that creates the version above.
+        type: string
+        required: true
+      build_matrix:
+        description: The build targets to produce as a JSON matrix.
+        type: string
+        required: true
+    secrets:
+      token:
+        description: The Github token or similar to authenticate with.
+        required: true
+      bucket:
+        description: The name of the S3 bucket to push packages into.
+        required: false
+      access_key_id:
+        description: The S3 access key id for the bucket.
+        required: false
+      secret_access_key:
+        description: The S3 secret access key for the bucket.
+        required: false
+
+jobs:
+  build-valkey:
+    # Capture source tarball and generate checksum for it
+    name: Build package ${{ matrix.distro.target }} ${{ matrix.distro.arch }}
+    runs-on: 'ubuntu-latest'
+    permissions:
```

Comment:
Consider restricting permissions on top-level instead of in the job (in case there is a new jobs added to this file).